pkg/acsengine/engine.go

```diff
@@ -1811,6 +1834,29 @@ write_files:
 	return fmt.Sprintf(clusterYamlFile, filelines)
 }
 
+// Identifies a distro to use for master parameters
+func getDistro(properties *api.Properties) api.Distro {
+	// Set a default distro
+	distro := api.Ubuntu
+
+	if properties.MasterProfile == nil {
+		// Check if AgentPoolProfiles have an assigned distro
+		// Use distro defined by first agent profile
+		if len(properties.AgentPoolProfiles) > 0 {
```

Comment:
Please only use the MasterProfile to determine the distro for master parameters. Properties.MasterProfile.Distro should be used to get the Distro. The default distro is set in setMasterNetworkDefaults.

Reply:
So how should I update [setMasterNetworkDefaults](https://github.com/Azure/acs-engine/blob/master/pkg/acsengine/defaults.go#L331) to look at the Agent pool profile? The reason I'm asking is because there's an api-model test related to [agents-only.json](https://github.com/Azure/acs-engine/blob/master/examples/agents-only.json).

Reply:
I think if the master is hosted (agent pool only) then the master distro should be Ubuntu by default. In any case, the master defaults should not be set by looking at the agents, @anhowe please correct me if I'm wrong.

Reply:
Maybe I don't understand the use case of agents only -- is it a special use case, what if a Master was CoreOS? I would assume, we'd want to match agents with the Master's OS?

Reply:
The agent-only flow supports AKS (managed kubernetes), and there is currently no support for using acs-engine as a driver for agent-only cluster workflows. So, generally, we will need to avoid modifying those code paths.

Reply:
@CecileRobertMichon The default distro that is set by  `setMasterNetworkDefaults` is not called by the main engine, it is hardcoded as `api.Ubuntu` in [getParameters](https://github.com/Azure/acs-engine/blob/master/pkg/acsengine/engine.go#L462-L465). I will modify this code to use MasterProfile.Distro.

Reply:
Because [setMasterNetworkDefaults](https://github.com/Azure/acs-engine/blob/master/pkg/acsengine/defaults.go#L325-L327) returns if the MasterProfile is nil, I will have to set the default distro as Ubuntu within the engine.